### PR TITLE
fix upsert cached object updater

### PR DESCRIPTION
**Pull request by earonesty**
found a bug in cvfs with upsert, need to ensure cache is not over-updated!

### CODE REVIEW

File: omen2/table.py [hunk 1]
```diff
@@ -122,6 +126,10 @@ def _add(self, obj: U, upsert: bool) -> U:
     def _notx_add(self, obj: U, upsert: bool = False) -> U:
         obj._bind(table=self)
         obj._commit(upsert=upsert)
+        if upsert:
+            with suppress(OmenNoPkError):
```
Comment (higleyc) [inline omen2/table.py:130]: could you add a comment on when this is expected

Comment (earonesty) [reply omen2/table.py:130]: i will write a test case.  it's really an edge case.   i think i should catch and clear the cache.

Comment (earonesty) [reply omen2/table.py]: yeah, it's impossible.  adding support for "no pk" situations is a bigger effort and not terribly useful

**Review by higleyc — APPROVED** [review]
